src/PKSim.Core/Snapshots/Mappers/DataInfoMapper.cs

```diff
@@ -20,18 +25,25 @@ public DataInfoMapper(ExtendedPropertiesMapper extendedPropertiesMapper)
          var snapshot = await SnapshotFrom(dataInfo, x =>
          {
             x.AuxiliaryType = dataInfo.AuxiliaryType;
-            x.Category = dataInfo.Category;
+            x.Category = SnapshotValueFor(dataInfo.Category);
             x.ComparisonThreshold = dataInfo.ComparisonThreshold;
             x.Date = dataInfo.Date;
             x.LLOQ = dataInfo.LLOQ;
-            x.MolWeight = dataInfo.MolWeight;
+            x.MolWeight = dataInfo.MolWeight.HasValue ? molWeightToDisplayValue(dataInfo) : null;
```

Comment:
remove the HasValue, you are already checking in the molWeightToDisplayValue
e.g. `x.MolWeight = molWeightToDisplayValue(dataInfo)`